includes/sanitizers/class-amp-form-sanitizer.php

```diff
@@ -150,16 +169,59 @@ public function sanitize() {
 	 * @param Element $form_element Form element.
 	 */
 	protected function normalize_target_attribute( Element $form_element ) {
-		$target = $form_element->getAttribute( 'target' );
+		$target = $form_element->getAttribute( Attribute::TARGET );
 		if ( '_top' !== $target ) {
 			if ( ! $target || in_array( $target, [ '_self', '_parent' ], true ) ) {
-				$form_element->setAttribute( 'target', '_top' );
+				$form_element->setAttribute( Attribute::TARGET, '_top' );
 			} elseif ( '_blank' !== $target ) {
-				$form_element->setAttribute( 'target', '_blank' );
+				$form_element->setAttribute( Attribute::TARGET, '_blank' );
 			}
 		}
 	}
 
+	/**
+	 * Normalize form action attribute.
+	 *
+	 * @param Element $form_element Form element.
+	 * @return string Normalized action URL.
+	 */
+	protected function normalize_action_attribute( Element $form_element ) {
+		$action_url = $this->make_http_url_schemeless(
+			$this->get_action_url( $form_element->getAttribute( Attribute::ACTION ) )
+		);
+		$form_element->setAttribute( Attribute::ACTION, $action_url );
+		return $action_url;
+	}
+
+	/**
+	 * Determine whether the form is for the comments.
+	 *
+	 * @param Element $form_element Form element.
+	 * @return bool Is comments form.
+	 */
+	protected function is_comments_form_element( Element $form_element ) {
+		return (
+			'commentform' === $form_element->getAttribute( Attribute::ID )
+			&&
+			'wp-comments-post.php' === basename( wp_parse_url( $form_element->getAttribute( Attribute::ACTION ), PHP_URL_PATH ) )
+		);
+	}
+
+	/**
+	 * Make HTTP URLs protocol-less, since HTTPS is required for forms.
+	 *
+	 * @todo If in loose sandboxing, we may want to allow non-HTTPS.
```

Comment:
Only case would be an endpoint elsewhere that is not available HTTPS. True this is increasingly uncommon, but it does happen. If we never get around to implementing it, I wouldn't be sad.

Reply:
Actually, it turns out that `make_http_url_schemeless` is dead code anyway, since the return value of `get_action_url()` is schemeless or HTTPS-enforced.

And if native post forms are allowed, then it passes through the `action` without any changes.

Reply:
Removed in 78684ae59c41a57ce21a32eb5c40bf458ac745fc